scrapyjs/middleware.py

```diff
@@ -96,20 +108,31 @@ def process_request(self, request, spider):
             # are not respected.
             headers=Headers({'Content-Type': 'application/json'}),
         )
-
         self.crawler.stats.inc_value('splash/%s/request_count' % endpoint)
         return req_rep
 
     def process_response(self, request, response, spider):
-        splash_options = request.meta.get("_splash_processed")
+        splash_options = self.get_splash_options(request, spider)
         if splash_options:
             endpoint = splash_options['endpoint']
             self.crawler.stats.inc_value(
                 'splash/%s/response_count/%s' % (endpoint, response.status)
             )
-
+            response = self.html_response(response, request)
```

Comment:
I also think that by default response should be returned without changes, but if we will have option to enable html response and use scrapyjs transparently i.e. getting rendered html without handling this in callback - that would be very, very convenient, e.g. you can use `response.css()`, `response.xpath()` and `response.body` right away  - just like if request was made to plain html page. 

I don't know if you already spent some time working on #12 and description doesn't provide much information. How it would work? You say it would be like FormRequest, but this has nothing to do with responses. Can you share a bit of your thoughts? One thing that came to mind is SplashRequest and SplashHtmlRequest classes where SplashRequest will return plain response from splash and SplashHtmlRequest will return rendered html along with headers and cookies returned by target site. But I'm not sure if my insight is correct.

If you insist - I think this part can be removed, we can override scrapyjs and add this inside the project. @pawelmhm ?